From 439f7d6b5435cde56c3183c0461f38387f4f1471 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 27 Sep 2018 12:48:30 +0300
Subject: [PATCH] EEprom Layout: update to new format

Update to v1 (no more legacy)
Update to 8B page EEprom boundary
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
index 9edf381..3046d68 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
@@ -34,18 +34,18 @@
 }
 
 LAYOUT_VER=`xxd -p -l1 ${LAYOUT}`
-if [ "${LAYOUT_VER}" -eq "0" ]; then
+if [ "${LAYOUT_VER}" -eq "1" ]; then
 	print_field_num "Layout Version" 0 1
-	print_field_str "Production Serial Number" 1 16
-	print_field_str "PCB Assembly Version" 17 4
-	print_field_str "PCB Version" 21 4
-	print_field_num "DDR Type and Speed" 25 2
-	print_field_num "PCIe Gen3 or Gen4 Enabled" 27 1
-	print_field_num "Max Frequency [MHz]" 28 2
-	print_field_num "Max Power [Watt]" 30 1
-	print_field_str "System Product" 31 8
-	print_field_str "Vendor System Product" 39 16
-	print_field_str "Board Description" 55 64
+	print_field_num "DDR Type and Speed" 1 2
+	print_field_num "PCIe Gen3 or Gen4 Enabled" 3 1
+	print_field_num "Max Frequency [MHz]" 4 2
+	print_field_num "Max Power [Watt]" 6 1
+	print_field_num "System Product HL" 7 1
+	print_field_str "Production Serial Number" 8 16
+	print_field_str "PCB Assembly Version" 24 4
+	print_field_str "PCB Version" 28 4
+	print_field_str "Vendor System Product" 32 16
+	print_field_str "Board Description" 48 64
 else
 	echo "Unknown eeprom layout"
 fi
diff --git a/board/habanalabs/goya/uboot-eeprom.cmd b/board/habanalabs/goya/uboot-eeprom.cmd
index 66579d5..fa23316 100644
--- a/board/habanalabs/goya/uboot-eeprom.cmd
+++ b/board/habanalabs/goya/uboot-eeprom.cmd
@@ -1,15 +1,15 @@
-setenv LAYOUT "-l legacy"
+setenv LAYOUT "-l v1"
 mw.b 7ff0000000 ff 100
 eeprom write  $LAYOUT 7ff0000000 0 100
-eeprom update $LAYOUT "Layout Version" "0"
-eeprom update $LAYOUT "Production Serial Number" "AI11223344"
-eeprom update $LAYOUT "PCB Assembly Version" "V1A"
-eeprom update $LAYOUT "PCB Version" "R0B"
+eeprom update $LAYOUT "Layout Version" "1"
 eeprom update $LAYOUT "DDR Type and Speed" "0314"
 eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "3"
 eeprom update $LAYOUT "Max Frequency [MHz]" "064a"
 eeprom update $LAYOUT "Max Power [Watt]" "c8"
-eeprom update $LAYOUT "System Product" "HL-102"
+eeprom update $LAYOUT "System Product HL" "66"
+eeprom update $LAYOUT "Production Serial Number" "AI11223344"
+eeprom update $LAYOUT "PCB Assembly Version" "V1A"
+eeprom update $LAYOUT "PCB Version" "R0B"
 eeprom update $LAYOUT "Vendor System Product" "F08GL0AIG000A"
 eeprom update $LAYOUT "Board Description" "Goya HL1000"
 echo "eeprom content is:"
